lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1629,6 +1661,8 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
     switchStmt->setSubjectExpr(subjectExpr);
     Type subjectType = switchStmt->getSubjectExpr()->getType();
 
+    diagnoseMoveOnlyPatternMatchSubject(getASTContext(), subjectExpr);
+
```

Comment:
Note you'll also need to cover the constraint system type-checking paths for these to handle them within closures, i.e `visitSwitchStmt` in `SyntacticElementSolutionApplication` and the `StmtCondition` branch in `ExprWalker::rewriteTarget`.